src/Lorentz/Contracts/BaseDAO/Doc.hs

```diff
@@ -0,0 +1,168 @@
+  If there is no such key, this entrypoint does nothing.
+
+  All tokens passed to this entrypoint must be unfrozen.
+  Each owner must be equal to sender, or the entrypoint fails.
+  |]
+
+transferOwnershipDoc :: Markdown
+transferOwnershipDoc  = [md|
+  Asks an Address to become an Admin. Can be called only by current Admin. The admin duties transfers only when the
+  requested address accepts ownership. If called multiple times, only the last called address can accept ownership.
+  |]
+
+acceptOwnershipDoc :: Markdown
+acceptOwnershipDoc = [md|
+  Accepts the administrator privelege.
+  Only works when sender was asked to become an admin and only if it was asked by current admin.
+  |]
+
+migrateDoc :: Markdown
+migrateDoc = [md|
+  Asks an address to migrate the contract to it.
+  Can be performed only by current administrator.
+  The contract is not considered migrated, until it recieves confirm_migration call.
+  |]
+
+confirmMigrationDoc :: Markdown
+confirmMigrationDoc = [md|
+  Confirms migration of a contract to the sender address.
+  After a successful call the contract will be set to migrated state where no operations are possible.
+  |]
+
+proposeDoc :: Markdown
+proposeDoc = [md|
+  Saves the proposal with specific id and freezes the ammount of sender tokens equal to the given ammount.
+  Sender must have enough unfrozen tokens.
+  The sender's ammount of frozen tokens is increased by proposal ammount. And the amount of unfrozen one
+  is decreased by the same value.
+  |]
+
+voteDoc :: Markdown
+voteDoc = [md|
+  For each vote params in a given list votes for the proposal with that id. Thus the sender can vote many proposal
+  (or one proposal multiple times) in a single call.
+  Sender must have an ammount required for all votings.
+  |]
+
+setVotingPeriodDoc :: Markdown
+setVotingPeriodDoc = [md|
+  Updates how long the voting period would last.
+  Can be performed only by administrator
+  It affects all ongoing proposals and all created afterwards.
+  |]
+
+setQuorumThresholdDoc :: Markdown
+setQuorumThresholdDoc = [md|
+  Updates the quorum threshold with a given value.
+  Can be performed only by administrator.
+  It affects all ongoing proposals and all created afterwards.
+  |]
+
+
+flushDoc :: Markdown
+flushDoc = [md|
+  Finish voting process on all proposals where the voting period is over.
+  Returns an ammount to the proposer, determined by the result of voting.
+  There is a possibility of some tokens being lost due to administrator perform
+  of burn or transfer operation.
+  If proposal is accepted, the decision lambda is called.
+  |]
+
+burnDoc :: Markdown
+burnDoc = [md|
+  Reduces the amount of tokens of the given address. Can be performed only by administrator
+  and only if the given address have enought tokens to burn.
+  |]
+
+mintDoc :: Markdown
+mintDoc = [md|
+  Provides the amount of tokens of the given address.
+  Can be performed only by administrator.
```

Comment:
Predicates like this are better put in code.

Could you remove this sentence everywhere and instead put `doc $ DRequireRole "admin"` into `authorizeAdmin`'s definition? If everything is done correctly, in all the respective places there should be a note about admin to be the only valid caller.

`DRequireRole` is defined in [morley-ledgers](https://gitlab.com/morley-framework/morley-ledgers/blob/master/code/morley-ledgers/src/Lorentz/Contracts/ManagedLedger/Doc.hs#L136) and probably can be just imported.

Reply:
Actually it is already there ...

Reply:
Oh really, memno.

Okay. That was really good that documentation mentioned about administrator explicitly, it's just that we can do this in a smarter way, fonrtunately.